src/screens/Voucher/VoucherScannerScreen.tsx

```diff
@@ -0,0 +1,69 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import { BarCodeScanningResult, Camera } from 'expo-camera';
+import * as Linking from 'expo-linking';
+import React, { useState } from 'react';
+import { Alert, StyleSheet } from 'react-native';
+
+import appJson from '../../../app.json';
+import { SafeAreaViewFlex } from '../../components';
+import { texts } from '../../config';
+import { ScreenName } from '../../types';
+import { QUERY_TYPES } from '../../queries';
+
+const parseQrCode = (
+  data: string
+): { query: string; queryVariables: { id: number | undefined } } => {
+  const result = Linking.parse(data);
+
+  if (result.scheme === appJson.expo.scheme && result.hostname === 'detail') {
```

Comment:
can we make the hostnames a constant somewhere in the configs section as we have more than one now? we should not let this grow hardcoded in several places.

Reply:
Yes, that's right. For this I will create a new section for `hostname` in `consts` and pull it from there. Like:
```
  HOST_NAMES: {
    DETAIL: 'detail',
    ENCOUNTER: 'encounter'
  },
```

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1064/commits/590d4d69adc0850e894de679de501a1ebf77f9d5